Sources/Misc/StoreKit2Setting.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2Setting.swift
+//
+//  Created by Nacho Soto on 4/6/22.
+
+/// Defines when StoreKit 2 APIs may be used
+enum StoreKit2Setting {
+
+    /// Never use SK2
+    case disabled
+
+    /// Use SK2 if available for certain APIs that provide a better implementation
+    case enabledOnlyForOptimizations
```

Comment:
This isn't used, yet, I'm guessing a follow up PR uses it.

Reply:
`enabledOnlyForOptimizations` is the default value (see `default` below)
It's the current existing behavior when `useStoreKit2IfAvailable` is `false`. We were only using SK2 when it made sense for better implementations.

The new `case` is `disabled`, which allows us to not even use SK2 in those cases.

The fact that you were confused here and in the other comment is probably because this name is bad :/ but @aboedo and I couldn't come up with a better one.

Hopefully with this explanation this make a bit more sense, and maybe you have a suggestion for how to make this behavior more clear?

Reply:
+1 to Nacho. FWIW the philosophy that this case applies is "use SK2 whenever its value is guaranteed to be more accurate than using SK1 because SK1 was abhorrent at doing this"

Reply:
which... we should probably elaborate on in this doc, including examples (which off the top of my head are: Intro eligibility, determining whether a receipt has any purchases on it, and manage subscriptions)